frontend/src/App.jsx

```diff
@@ -0,0 +1,51 @@
+// App.js
+import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import Home from './components/Home';
+import About from './components/About';
+import BookList from './components/BookList';
+import NewBook from './components/NewBook';
+import Signup from './components/Signup';
+import Login from './components/Login';
+import Header from './components/Header';
+import CurrentBook from './components/CurrentBook';
+
+const App = () => {
+  const [isUserLoggedIn, setUserLoggedIn] = useState(false);
+  const [currentBook, setCurrentBook] = useState(null);
+
+  useEffect(() => {
+    const isLoggedIn = localStorage.getItem('isLoggedIn') === 'true';
+    setUserLoggedIn(isLoggedIn);
+  }, []);
+
+  const addCurrentBook = (book) => {
+    setCurrentBook(book);
+  };
+
+  return (
+    <BrowserRouter>
+      {isUserLoggedIn ? (
+        <>
+          <Header setUserLoggedIn={setUserLoggedIn} />
```

Comment:
@omkogde-vf How are we handling refresh event?, as i guess on refresh user is logged out.

Reply:
Now I have handled it by putting the token true in localstorage by using localStorage.setItem("isLoggedIn", "true");